phlib/mapimg.c

```diff
@@ -2371,3 +2469,98 @@ BOOLEAN PhGetMappedImageDebugEntryByType(
 
     return FALSE;
 }
+
+NTSTATUS PhGetMappedImageEhCont32(
+    _Out_ PPH_MAPPED_IMAGE_EH_CONT EhContConfig,
+    _In_ PPH_MAPPED_IMAGE MappedImage
+)
+{
+    NTSTATUS status;
+    PIMAGE_LOAD_CONFIG_DIRECTORY32 config32;
+
+    if (!NT_SUCCESS(status = PhGetMappedImageLoadConfig32(MappedImage, &config32)))
+        return status;
+
+    // Not every load configuration contains eh continuation
+    if (!RTL_CONTAINS_FIELD(config32, config32->Size, GuardEHContinuationCount))
+        return STATUS_INVALID_VIEW_SIZE;
+
+    EhContConfig->EhContTable = PhMappedImageRvaToVa(MappedImage, (ULONG)(config32->GuardEHContinuationTable - MappedImage->NtHeaders->OptionalHeader.ImageBase), NULL);
```

Comment:
This needs to be MappedImage->NtHeaders32 for the 32bit headers instead of MappedImage->NtHeaders 

Optionally that statement could also be: PtrToUlong(PTR_SUB_OFFSET(config32->GuardEHContinuationTable, MappedImage->NtHeaders32->OptionalHeader.ImageBase)) but that's entirely personal style and optional.

Reply:
Nice catch, EHCont doesn't seem to be supported on x86, so there is no way to test this branch, I added it just in case MS adds support for in the future, which I doubt they will since exceptions work in a different way on x86.

Reply:
No, I used same compiler (VS 16.7.2) on different projects, but one was EXE (size of entry was 4, so just RVA) and one was DLL (size 5).

Reply:
The last comment was incorrect and I removed it. The change in size might be the actual flag for determining if the module is an EXE vs a DLL.

MSVC also currently uses two different versions of the same compiler and can be seen in the RICH metadata (ProdID) tab in PEViewer so it could also be an older version creating the size difference... I guess we'll know eventually what causes it.